lib/src/model.dart

```diff
@@ -1326,6 +1350,28 @@ class Extension extends Container
         ElementType.from(_extension.extendedType, library, packageGraph);
   }
 
+  /// Returns [true] if this extension could be applicable to any possible
+  /// instantiation of [c].
+  bool couldApplyTo(Class c) =>
+      _couldApplyTo(extendedType.type, c.element, packageGraph.typeSystem);
+
+  static bool _couldApplyTo(
+      DartType extendedType, ClassElement element, Dart2TypeSystem typeSystem) {
+    InterfaceTypeImpl classInstantiated =
+        typeSystem.instantiateToBounds(element.thisType);
+    classInstantiated = element.instantiate(
+        typeArguments: classInstantiated.typeArguments.map((a) {
+          if (a.isDynamic) {
+            return typeSystem.typeProvider.neverType;
+          }
+          return a;
+        }).toList(),
+        nullabilitySuffix: classInstantiated.nullabilitySuffix);
```

Comment:
Let's be explicit, use `NullabilitySuffix.none`.

Reply:
I would prefer to preserve whatever I was given rather than explicitly specify something wrong.